### PR TITLE
2 rooms with separation door

### PR DESCRIPTION
**Pull request by jlurgo**

### CODE REVIEW

File: ca_gazebo/launch/ekuthon_coop_robots.launch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<launch>
+  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
+  <include file="$(find gazebo_ros)/launch/empty_world.launch">
+    <arg name="world_name" value="$(find ca_gazebo)/worlds/two_rooms.world"/>
+    <!-- more default parameters can be changed here -->
+  </include>
+  <!-- Spawn robot/s -->
+  <param name="create1/amcl/initial_pose_x" value="0"/>
```
Comment (eborghi10) [inline ca_gazebo/launch/ekuthon_coop_robots.launch]: Add also parameters for a second robot.

File: ca_gazebo/worlds/two_rooms.world [hunk 2]
```diff
@@ -0,0 +1,1132 @@
...
+                <soft_cfm>0</soft_cfm>
+                <soft_erp>0.2</soft_erp>
+                <kp>1e+13</kp>
+                <kd>1</kd>
+              </bullet>
+            </contact>
+          </surface>
+        </collision>
+      </link>
+      <link name='door'>
```
Comment (eborghi10) [inline ca_gazebo/worlds/two_rooms.world]: Did you try adding the door as a separated model? I recommend you another way:

- The joint of the door, instead of attaching the joint to the wall, add it to the ground.

File: ca_gazebo/worlds/two_rooms.world [hunk 3]
```diff
@@ -0,0 +1,1132 @@
...
+        <inertial>
+          <mass>0.05</mass>
```
Comment (eborghi10) [inline ca_gazebo/worlds/two_rooms.world]: Don't forget to increase the mass.

**Review by eborghi10 — CHANGES_REQUESTED** [review]
I left some comments, PTAL.

File: ca_gazebo/src/model_pose_publisher_plugin.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      // Store the pointer to the model
+      this->model = _parent;
+
+      // Listen to the update event. This event is broadcast every
+      // simulation iteration.
+      this->updateConnection = event::Events::ConnectWorldUpdateBegin(
+          std::bind(&ModelPosePublisherPlugin::OnUpdate, this));
+      this->prev_update_time_ = ros::Time::now();
+
+      // this->rosnode_.reset(new ros::NodeHandle("/model_pose_plugin"));
```
Comment (eborghi10) [inline ca_gazebo/src/model_pose_publisher_plugin.cpp]: Do you need the code commented here?

File: ca_gazebo/src/model_pose_publisher_plugin.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      // this->pose_pub_ = this->rosnode_->advertise<std_msgs::String>("model_pose", 1000);
+    }
+    // Called by the world update start event
+    public: void OnUpdate()
+    {
+      if ((ros::Time::now() - this->prev_update_time_) < update_rate) {
+       return;
+     }
+     ROS_INFO("publishing door pose");
```
Comment (eborghi10) [inline ca_gazebo/src/model_pose_publisher_plugin.cpp]: This code is door-agnostic. You can do something like this:
```C++
ROS_DEBUG_STREAM_NAMED("Pose Publisher plugin",
   this->model pose << ":\t" <<
   this->model->RelativePose().Pos().X() << " " <<
   this->model->RelativePose().Pos().Y() << " " <<
   this->model->RelativePose().Pos().Z());
```

Comment (eborghi10) [reply ca_gazebo/src/model_pose_publisher_plugin.cpp]: See that I used `DEBUG` instead of `INFO`.

**Review by eborghi10 — CHANGES_REQUESTED** [review]
Some minor comments, it's looking good!

**Comment by jlurgo** [comment]
All changes pushed @eborghi10

File: ca_gazebo/launch/create_empty_world.launch [hunk 6]
```diff
@@ -1,20 +1,21 @@
 <launch>
 	<arg name="env" default="empty"/>
-
-  	<arg name="paused" value="$(optenv PAUSED false)"/>
+	<arg name="paused" value="$(optenv PAUSED false)"/>
```
Comment (eborghi10) [inline ca_gazebo/launch/create_empty_world.launch:5]: Revert this change, please.

File: ca_gazebo/launch/ekuthon_coop_robots.launch [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<launch>
+  <param name="create1/amcl/initial_pose_x" value="0.29"/>
```
Comment (eborghi10) [inline ca_gazebo/launch/ekuthon_coop_robots.launch]: Using the old values would be the best:

```
<param name="create1/amcl/initial_pose_x" value="-1"/>
<param name="create1/amcl/initial_pose_y" value="1"/>
<param name="create1/amcl/initial_pose_a" value="0"/>

<param name="create2/amcl/initial_pose_x" value="1"/>
<param name="create2/amcl/initial_pose_y" value="1"/>
<param name="create2/amcl/initial_pose_a" value="3.1416"/>
```

File: ca_gazebo/launch/ekuthon_coop_robots.launch [hunk 8]
```diff
@@ -0,0 +1,13 @@
+<launch>
```
Comment (eborghi10) [inline ca_gazebo/launch/ekuthon_coop_robots.launch]: Maintain the naming convention of the rest of the launch files. Call this `create_cooperative_two_rooms` for example.

**Review by eborghi10 — CHANGES_REQUESTED** [review]
3 more minor comments.

Also, please delete the [WIP] in the title since it's working now. :+1:

**Review by eborghi10 — APPROVED** [review]
Awesome, thanks @jlurgo !!
